src/python/pants/bin/local_pants_runner.py

```diff
@@ -221,11 +221,22 @@ def set_start_time(self, start_time):
     if self._repro:
       self._repro.capture(self._run_tracker.run_info.get_as_dict())
 
+  def _set_exiter(self, set_global_exiter=False):
     # The __call__ method of the Exiter allows for the prototype pattern.
     self._exiter = LocalExiter(self._run_tracker, self._repro, exiter=self._exiter)
-    ExceptionSink.reset_exiter(self._exiter)
+    if set_global_exiter:
+      ExceptionSink.reset_exiter(self._exiter)
+
+  def run(self, start_time=None, set_global_exiter=True):
```

Comment:
Ditto. Although I'm not sure what `start_time` is.. `Optional[float]`? `Optional[time.time]`?